src/he/wolf.c

```diff
@@ -147,21 +165,21 @@ int he_wolf_dtls_write(WOLFSSL *ssl, char *buf, int sz, void *ctx) {
   }
 
   // Call the write callback if set
-  if(conn->outside_write_cb) {
-    res = conn->outside_write_cb(conn, conn->write_buffer, post_plugin_length, conn->data);
+  if(conn->outside_write_cb || conn->outside_write_ex_cb) {
```

Comment:
I think, we could simplify this a bit by not checking `conn->outside_write_cb || conn->outside_write_ex_cb` here and simply call `he_internal_conn_outside_write` with the buffer.

`he_internal_conn_outside_write` can send or ignore based on 
`conn->outside_write_cb || conn->outside_write_ex_cb` by returning success

instead of returning failure on `(conn->outside_write_cb == NULL && conn->outside_write_ex_cb == NULL) `


example:
```c
static he_return_code_t he_internal_conn_outside_write(he_conn_t *conn, uint8_t *packet,
                                                       size_t length) {
  if(conn == NULL || packet == NULL) {
    return HE_ERR_NULL_POINTER;
  }
  if(conn->outside_write_cb == NULL || conn->outside_write_ex_cb == NULL) {
    return HE_SUCCESS;
  }

...
}
```

Reply:
I considered it yesterday, but I feel this doesn't simplify the logic much, but it would cause `he_internal_conn_outside_write` being called 3 times unnecessarily.

Reply:
I don't the understand the benefit of this approach. 
Cause there will not be any usable connection, if there is no `outside_write_cb` or `outside_write_ex_cb` set. 
I believe it is not a valid case. The NULL check is only to make sure we don't do NULL pointer reference.

So in any usable connection, we will call this function for one time in normal case and three times if aggressive mode is enabled.

Please correct me if my understanding is wrong.

Reply:
You're right, we can remove the `if(conn->outside_write_cb || conn->outside_write_ex_cb)` checks here. But I think he_internal_conn_outside_write should still return err when both callbacks are NULL. It's a fatal error, we shouldn't hide it.